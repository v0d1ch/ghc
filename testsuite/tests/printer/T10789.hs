{-# LANGUAGE TypeFamilies, PolyKinds, UndecidableInstances #-}

module T10789 where

import Data.Proxy

type family F (a :: k) :: k
type instance F a = G a

type family G a
type instance G a = a

foo :: Proxy (F Maybe) -> Proxy Maybe
foo = id
