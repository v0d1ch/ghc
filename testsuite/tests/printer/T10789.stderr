T10789.hs:14:7:
    Couldn't match type ‘F Maybe’ with ‘Maybe’
    Expected type: Proxy (F Maybe) -> Proxy Maybe
      Actual type: Proxy Maybe -> Proxy Maybe
    In the expression: id
    In an equation for ‘foo’: foo = id
Failed, modules loaded: none.
